ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -848,8 +955,366 @@ void adjust_conv_op_config_for_auto_shard_if_necessary(
+        uint32_t weight_block_num_tiles = weight_block_w_ntiles * act_block_w_ntiles; //act_block_w_ntiles == weight_block_h_ntiles
+        uint32_t weight_block_num_bytes = weight_block_num_tiles * weights_tile_size;
+
+        uint32_t bias_block_num_bytes = per_core_out_matrix_width_ntiles * bias_tile_size;
+
+        uint32_t out_block_num_tiles = per_core_out_matrix_height_ntiles * per_core_out_matrix_width_ntiles;
+
+        uint32_t num_blocks_act_w = weight_matrix_height_ntiles / act_block_w_ntiles;
+
+        packer_l1_acc = packer_l1_acc && ((enable_bias && num_blocks_act_w > 1) || (num_blocks_act_w > 2));
+
+        auto interm_dtype =
+            packer_l1_acc ? (fp32_dest_acc_en ? DataType::FLOAT32 : DataType::BFLOAT16) : output_dtype;
+
+        uint32_t partial_tile_size = tt::tile_size(datatype_to_dataformat_converter(interm_dtype));
+
+        uint32_t partials_block_num_bytes = out_block_num_tiles * partial_tile_size;
+
+
+        //CB 0
+        uint32_t cb0_size = tilized_act_block_num_bytes; tt::log_debug(tt::LogOp, "CB0 Size: {}", cb0_size);
+
+        //CB 1
+        uint32_t cb1_size = weight_block_num_bytes; tt::log_debug(tt::LogOp, "CB1 Size: {}", cb1_size);
+
+        //CB 2
+        uint32_t cb2_size = bias_block_num_bytes; tt::log_debug(tt::LogOp, "CB2 Size: {}", cb2_size);
+
+        //CB 5
+        uint32_t cb5_size = 64; tt::log_debug(tt::LogOp, "CB5 Size: {}", cb5_size);
+
+        //CB 6
+        uint32_t cb6_size = act_block_num_bytes; tt::log_debug(tt::LogOp, "CB6 Size: {}", cb6_size);
+
+        //CB 24
+        uint32_t cb24_size = partials_block_num_bytes;
+        if(interm_dtype == output_dtype) {
+            cb24_size = 0;
+        } else {
+            tt::log_debug(tt::LogOp, "CB24 Size: {}", cb24_size);
+        }
+        //CB 25
+        uint32_t cb25_size = tilized_act_block_num_bytes; tt::log_debug(tt::LogOp, "CB25 Size: {}", cb25_size);
+
+        uint32_t total_CB_size = cb0_size + cb1_size + cb2_size + cb5_size + cb6_size + cb24_size + cb25_size;
+
+        tt::log_debug(tt::LogOp, "Total CB Size: {}", total_CB_size);
+
+        return {output_size_per_core_in_bytes, total_CB_size};
+    } else if (shard_layout == TensorMemoryLayout::HEIGHT_SHARDED) {
+        uint32_t output_size = 0;
+        if(use_non_tile_height){
+            uint32_t total_height = conv_output_h * conv_output_w * batch_size;
+            output_size = total_height / pconfig.num_cores_nhw * output_channels;
+        } else {
+            output_size = per_core_out_matrix_height_ntiles *
+                          per_core_out_matrix_width_ntiles *
+                          output_tile_size;
+        }
+
+        uint32_t bias_block_num_bytes = per_core_out_matrix_width_ntiles * bias_tile_size;
+
+        uint32_t conv_act_c_blocks = weight_matrix_width_ntiles / per_core_out_matrix_width_ntiles;
+
+        TT_FATAL(conv_act_c_blocks == 1, "Error: conv_act_c_blocks should be 1 for height sharding");
```

Comment:
you already have this assert in the factory, no need for some random set of asserts here as well

Reply:
Removed.